packages/twenty-server/src/engine/core-modules/domain-manager/service/domain-manager.service.ts

```diff
@@ -21,10 +21,28 @@ export class DomainManagerService {
     private readonly environmentService: EnvironmentService,
   ) {}
 
-  getBaseUrl() {
-    const baseUrl = new URL(
-      `${this.environmentService.get('FRONT_PROTOCOL')}://${this.environmentService.get('FRONT_DOMAIN')}`,
-    );
+  getFrontUrl() {
+    let baseUrl: URL;
+
+    if (!this.environmentService.get('FRONT_DOMAIN')) {
+      baseUrl = new URL(this.environmentService.get('SERVER_URL'));
+    } else {
+      baseUrl = new URL(
+        `${this.environmentService.get('FRONT_PROTOCOL')}://${this.environmentService.get('FRONT_DOMAIN')}`,
+      );
+
+      const port = this.environmentService.get('FRONT_PORT');
+
+      if (port) {
+        baseUrl.port = port.toString();
+      }
+    }
+
+    return baseUrl;
+  }
```

Comment:
The PORT of the SERVER_URL should differ from the frontend.

We should allow to set the FRONT_PORT without the FRONT_PROTOCOL and FRONT_DOMAIN.

Like that:

```suggestion
  getFrontUrl() {
    let baseUrl: URL;

    if (!this.environmentService.get('FRONT_DOMAIN')) {
      baseUrl = new URL(this.environmentService.get('SERVER_URL'));
    } else {
      baseUrl = new URL(
        `${this.environmentService.get('FRONT_PROTOCOL')}://${this.environmentService.get('FRONT_DOMAIN')}`,
      );

      
    }
const port = this.environmentService.get('FRONT_PORT');

      if (port) {
        baseUrl.port = port.toString();
      }
    return baseUrl;
  }
```

Reply:
Our main concern is to lower the number of env variables needed
- in self hosting main case: we want front to be served on same url as server and we only have to specify SERVER_URL
- in local development case: we only have to specify FRONT_DOMAIN and FRONT_PORT in .env

Actually, I don't disagree with you, we could change port and protocol in separate if conditions and achieve the same goal

@Weiko what do you think?